INV4/pallet-ipl/src/tests.rs

```diff
@@ -0,0 +1,248 @@
+            0,
+            Default::default(),
+            [0, 0],
+            true,
+        ));
+
+        assert_eq!(Permissions::<Runtime>::get((0, 0), [0, 0]), Some(true));
+    });
+}
+
+#[test]
+fn set_permission_should_fail() {
+    ExtBuilder::default().build().execute_with(|| {
+        Ipl::create(
+            0,
+            InvArchLicenses::GPLv3,
+            percent!(50),
+            OneOrPercent::One,
+            false,
+        );
+
+        // Case 0: Unsigned origin
+        assert_noop!(
+            Ipl::set_permission(Origin::none(), 0, 0, [0, 0], true),
+            DispatchError::BadOrigin
+        );
+
+        // Case 1: Non-multisig origin
+        assert_noop!(
+            Ipl::set_permission(Origin::signed(ALICE), 0, 0, [0, 0], true),
+            Error::<Runtime>::NoPermission
+        );
+
+        // Case 2: Ipl does not exist
+        assert_noop!(
+            Ipl::set_permission(
+                Origin::signed(multi_account_id::<Runtime, IplId>(0, None)),
+                32,
+                0,
+                [0, 0],
+                true
+            ),
+            Error::<Runtime>::IplDoesntExist
+        );
+
+        assert_eq!(Permissions::<Runtime>::get((0, 0), [0, 0]), None);
+    });
+}
+
+#[test]
+fn set_asset_weight_should_work() {
+    ExtBuilder::default().build().execute_with(|| {
+        Ipl::create(
+            0,
+            InvArchLicenses::GPLv3,
+            percent!(50),
+            OneOrPercent::One,
+            false,
+        );
+
+        assert_ok!(Ipl::set_asset_weight(
+            Origin::signed(multi_account_id::<Runtime, IplId>(0, None)),
+            0,
+            0,
+            percent!(30),
+        ));
+
+        assert_eq!(AssetWeight::<Runtime>::get(0, 0), Some(percent!(30)));
+    });
+}
+
+#[test]
+fn set_asset_weight_should_fail() {
+    ExtBuilder::default().build().execute_with(|| {
+        Ipl::create(
+            0,
+            InvArchLicenses::GPLv3,
+            percent!(50),
+            OneOrPercent::One,
+            false,
+        );
+
+        // Case 0: Unsigned origin
+        assert_noop!(
+            Ipl::set_asset_weight(Origin::none(), 0, 0, percent!(50)),
+            DispatchError::BadOrigin,
+        );
+
+        // Case 1: Non-multisig origin
+        assert_noop!(
+            Ipl::set_asset_weight(Origin::signed(BOB), 0, 0, percent!(50)),
+            Error::<Runtime>::NoPermission,
+        );
+
+        // Case 2: Ipl does not exist
+        assert_noop!(
+            Ipl::set_asset_weight(
+                Origin::signed(multi_account_id::<Runtime, IplId>(0, None)),
+                32767,
+                0,
+                percent!(50)
+            ),
+            Error::<Runtime>::IplDoesntExist
+        );
+
+        assert_eq!(AssetWeight::<Runtime>::get(0, 0), None);
+    });
+}
+
+// Test does not include "should_fail" since it's not meant to fail
+#[test]
+#[ignore]
```

Comment:
Ignoring this test for now, since unwrapping is used [here](https://github.com/InvArch/InvArch-Frames/blob/main/INV4/pallet-ipl/src/lib.rs#L186)